another attempt to fix ToXContent

mkhludnev opened the pull request:
I'm trying to guess logic behind XContent fragments, but I rather think it's worth to check with someone who better understand it.

mkhludnev committed:
another attempt to fix for "Can not start an object, expecting field name (context: Object)" honoring XContent/Object/Fragment markers. In doubts.

Signed-off-by: Mikhail Khludnev <mkhl@apache.org>

mkhludnev commented:
@macohen @epugh ping

macohen commented on src/main/java/com/o19s/es/ltr/feature/store/StoredFeatureSet.java:
```diff
@@ -44,7 +45,7 @@
 
-public class StoredFeatureSet implements FeatureSet, Accountable, StorableElement {
+public class StoredFeatureSet implements FeatureSet, Accountable, StorableElement, ToXContentObject {
     public static final int MAX_FEATURES = 10000;
```
StorableElement extends ToXContent; maybe it should also implement ToXContentObject?

mkhludnev replied:
just did that in the last commit, also had to tweak tests

macohen commented on src/main/java/com/o19s/es/ltr/logging/LoggingSearchExtBuilder.java:
```diff
@@ -77,9 +77,11 @@ public static LoggingSearchExtBuilder parse(XContentParser parser) throws IOExce
+        // embraces fragment
+        return  builder.field(LOG_SPECS.getPreferredName(), logSpecs);
+        //return builder.endObject();
     }
```
not sure if this is always a fragment. I also don't think I trust the test suite to tell us. line coverage is at 45% in com.o19.es.

mkhludnev replied:
`logSpecs` is a list, so it's hardly might be a fragment 
`LoggingSearchExtBuilder.LogSpec implements ToXContentObject` and has ` builder.startObject();` in `toXContent()` so. they are not fragments.

macohen commented:
I'd love to get some more eyes on this. @noCharger? @msfroh mentioned he could take a look...

macohen commented:
Oh; and thanks so much for checking this out @mkhludnev!

mkhludnev committed:
StorableElement descendants are all ToXContentObject.

StoredFeatureParserTests.testToXContent() does what StoredFeatureSetParserTests do.

Signed-off-by: Mikhail Khludnev <mkhl@apache.org>

mkhludnev commented on src/test/java/com/o19s/es/ltr/feature/store/StoredFeatureParserTests.java:
```diff
@@ -98,7 +98,8 @@ public void testToXContent() throws IOException {
         XContentBuilder builder = MediaTypeRegistry.contentBuilder(XContentType.JSON);
-        featureString = String.valueOf(feature.toXContent(builder, ToXContent.EMPTY_PARAMS));
+        //featureString = String.valueOf(feature.toXContent(builder, ToXContent.EMPTY_PARAMS));
+        featureString = Strings.toString(XContentType.JSON, feature);//.toXContent(builder, ToXContent.EMPTY_PARAMS));
```
@macohen align test with `StoredFeatureSetParserTests.testToContent()`

mkhludnev commented on src/test/java/com/o19s/es/ltr/feature/store/StoredFeatureSetParserTests.java:
```diff
@@ -88,8 +88,6 @@ public void testToXContent() throws IOException {
         String featureSetString = generateRandomFeatureSet("my_set", features::add);
         StoredFeatureSet featureSet = parse(featureSetString);
 
-        XContentBuilder builder = XContentFactory.jsonBuilder();//contentBuilder(XContentType.JSON);
```
no use
